File: 03_Basic_Recursion/08_String_Palindrome.cs
Console.Write("Enter a string: ");
string input = Console.ReadLine();

Console.WriteLine($"Is the string a palindrome? : {f(input, 0, input.Length - 1)}");

bool f(string s,int l, int r)
{
    if(l>=r) return true;
    if(s[l] != s[r]) return false;

    return f(s, l+1, r-1);
}

File: 03_Basic_Recursion/08_String_Palindrome.cpp
#include <bits/stdc++.h>
using namespace std;

bool f(string s, int l, int r)
{
    if (l >= r)
        return true;

    if (s[l] != s[r])
        return false;

    return f(s, l + 1, r - 1);
}

int main()
{
    string input;

    cout << "Enter a string: ";
    cin >> input;

    cout << "Is the string a palindrome? : "
         << (f(input, 0, input.length() - 1) ? "true" : "false")
         << endl;

    return 0;
}